src/app/players/playerpoints.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { AngularFirestore, AngularFirestoreCollection } from '@angular/fire/firestore';
+import { map } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PlayerPointsService {
+
+  playerPointsCollection: AngularFirestoreCollection<PlayerPoint>;
+
+  constructor(private afs: AngularFirestore) {
+    this.playerPointsCollection = this.afs.collection('playerPoints');
+  }
+
+  getPlayerPoint(uid: string, seasonId: string): Observable<PlayerPoint> {
+    const playerPoint = this.afs.collection('playerPoints', ref =>
+      ref.where('seasonId', '==', seasonId));
+
+    return playerPoint.doc<PlayerPoint>(uid).valueChanges();
+  }
+
+  getTotalPlayerPoints(uid: string): Observable<PlayerPoint[]> {
+    const playerPoints: AngularFirestoreCollection<PlayerPoint> = this.afs.collection('playerPoints', ref =>
+      ref.where('playerId', '==', uid));
+
+    return playerPoints.snapshotChanges().pipe(
+      map((playerPts) => playerPts.map(item => item.payload.doc.data()))
+    );
+  }
+
+  getSeasonExp(uid: string, seasonId: string) {
+    this.getPlayerPoint(uid, seasonId).pipe(
+      map((playerPoint: PlayerPoint) => playerPoint ? playerPoint.totalPoints : null)
+    );
+  }
+
+  getTotalExp(uid: string): Observable<number> {
+    return this.getTotalPlayerPoints(uid).pipe(map((playerPoints) => {
+      let totalPoints = 0;
+      playerPoints.forEach(playerPoint => {
```

Comment:
array.reduce is better suited for this solution

Reply:
Not sure if I can/how to use reduce on an array of playerpoints instead of an array of numbers tho

Reply:
Follow [Sum_of_values_in_an_object_array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#Sum_of_values_in_an_object_array)